src/main/scala/net/snowflake/spark/snowflake/SnowflakeTelemetry.scala

```diff
@@ -91,6 +91,40 @@ object SnowflakeTelemetry {
       case _ => None
     }
 
+  /**
+    * Put the pushdown failure telemetry message to internal cache.
+    * The message will be sent later in batch.
+    *
+    * @param plan The logical plan to include the unsupported operations
+    * @param exception The pushdown unsupported exception
+    */
+  def addPushdownFailMessage(plan: LogicalPlan,
+                             exception: SnowflakePushdownUnsupportedException)
```

Comment:
be aware of logical plan telemetry sometime fails to send because the String is too long.

Reply:
The telemetry messages are collected for statistics purpose. It doesn't matter if a small amount of messages are not sent successfully, I think.